torch/csrc/inductor/aoti_torch/tensor_converter.cpp

```diff
@@ -0,0 +1,39 @@
+
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+#include <torch/csrc/inductor/aoti_torch/tensor_converter.h>
+
+namespace torch {
+namespace aot_inductor {
+
+at::Tensor* tensor_handle_to_tensor_pointer(AtenTensorHandle handle) {
+  return reinterpret_cast<at::Tensor*>(handle);
+}
+
+AtenTensorHandle tensor_pointer_to_tensor_handle(at::Tensor* tensor) {
+  return reinterpret_cast<AtenTensorHandle>(tensor);
+}
+
+std::vector<AtenTensorHandle> borrow_tensors_to_handles(
+    std::vector<at::Tensor>& tensors) {
+  std::vector<AtenTensorHandle> result;
+  result.reserve(tensors.size());
+  for (auto tensor : tensors) {
+    auto allocated = new at::Tensor(std::move(tensor));
+    result.push_back(tensor_pointer_to_tensor_handle(allocated));
+  }
+  return result;
+}
+
+std::vector<at::Tensor> borrow_handles_to_tensors(
+    std::vector<AtenTensorHandle>& handles) {
+  std::vector<at::Tensor> result;
+  result.reserve(handles.size());
+  for (auto handle : handles) {
+    auto tensor = *tensor_handle_to_tensor_pointer(handle);
```

Comment:
this is what triggered the refcount bump, fyi

Reply:
Yes, and calling `aoti_torch_delete_tensor_object` on `handle` will dec refcount. Because I am not doing that, I call it borrow here. I think the borrow concept is on the storage ptr, not on tensor object, which is hard to do when crossing C boundary.

Reply:
This comment doesn't make sense to me.

If you increment refcount and don't have a matching decrement somewhere else, that's a memory leak. If you increment refcount, and then expect the caller/callee to handle the decrement, that's a steal. Nowhere does "borrow" figure here.